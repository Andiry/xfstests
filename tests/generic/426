#! /bin/bash
# FS QA Test No. 426
#
# Check stale handles pointing to unlinked files
# and non-stale handles pointing to linked files
#
#-----------------------------------------------------------------------
# Copyright (C) 2017 CTERA Networks. All Rights Reserved.
# Author: Amir Goldstein <amir73il@gmail.com>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter

# real QA test starts here

# Modify as appropriate.
_supported_fs generic
_supported_os Linux
_require_test
# _require_exportfs already requires open_by_handle, but let's not count on it
_require_test_program "open_by_handle"
_require_exportfs

numfiles=1024
testdir=$TEST_DIR/$seq-dir
mkdir -p $testdir

# Check stale handles to deleted files
rm -f $testdir/*
src/open_by_handle -c $testdir $numfiles
src/open_by_handle -d $testdir $numfiles

# Check non-stale handles to linked files
rm -f $testdir/*
src/open_by_handle -c $testdir $numfiles
src/open_by_handle    $testdir $numfiles

# Check non-stale handles to files that were hardlinked and original deleted
rm -f $testdir/*
src/open_by_handle -c $testdir $numfiles
src/open_by_handle -l $testdir $numfiles
src/open_by_handle -u $testdir $numfiles

echo "Silence is golden"
status=0
exit
