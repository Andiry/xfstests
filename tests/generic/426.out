QA output created by 426
Silence is golden
